### DISCUSSION TITLE
cxx-qt-lib: Can we use C++ templates to reduce the amount of C++ code

### DISCUSSION THREAD

**Discussion by LeonMatthes** [KDAB/cxx-qt#370] [Ideas]
The most common C++ functions we need for cxx-qt-lib are:
- Various constructors (default, copy, etc.)
- Destructor

So could we just add a "cxx-qt-lib/common-utils.h" that defines templates for these.
e.g. for the destructor:
``` cpp
template<class T>
void destruct(T &t) {
    t.~T();
}
```

We've already found that CXX can deal with this pretty well, as long as it can infer all template type arguments.

**Comment by ahayzen-kdab** [comment]
We think yes as this already is done with `QSet<T>`.
